src/handler/mod.rs

```diff
@@ -580,36 +571,18 @@ impl Handler {
     ) {
         // Check that this challenge matches a known active request.
         // If this message passes all the requisite checks, a request call is returned.
-        let mut request_call = {
-            // Check for an active request
-            let node_address = match self.active_requests_nonce_mapping.remove(&request_nonce) {
-                Some(addr) => addr,
-                None => {
-                    trace!("Received a WHOAREYOU packet that references an unknown or expired request. Source {}, message_nonce {}", src_address, hex::encode(request_nonce));
+        let mut request_call = match self.active_requests.remove_by_nonce(&request_nonce) {
+            Some((node_address, request_call)) => {
+                // Verify that the src_addresses match
+                if node_address.socket_addr != src_address {
+                    trace!("Received a WHOAREYOU packet for a message with a non-expected source. Source {}, expected_source: {} message_nonce {}", src_address, node_address.socket_addr, hex::encode(request_nonce));
```

Comment:
I think some logic has changed here. 

In this case, we want to add the mapping back. This will prevent other random people from tampering with peoples requests if they know the nonce etc.

Reply:
Yes, I see it. Fixing it now.